src/oscar/apps/basket/forms.py

```diff
@@ -9,10 +9,38 @@
 
 Line = get_model('basket', 'line')
 Basket = get_model('basket', 'basket')
+Option = get_model('catalogue', 'option')
 Product = get_model('catalogue', 'product')
 
 
+def _option_text_field(option):
```

Comment:
Can I suggest a slightly different API here, something like this, in a separate module:

```
class OptionFieldRenderer:
    field_class = forms.CharField

    def __init__(self, product, option):
        self.product = product
        self.option = option

    def get_field_class(self):
        return self.field_class
    
    def get_field_kwargs(self):
        return {
	    'label': self.option.name,
            'required': self.option.required
        }

    def get_form_field(self):
        field_class = self.get_field_class()
        return field_class(self.get_field_kwargs())


class IntegerOptionFieldRenderer(OptionFieldRenderer):
    field_class = forms.IntegerField


class BooleanOptionFieldRenderer(OptionFieldRenderer):
    field_class = forms.BooleanField
```

I think this would be more flexible because (a) it allows projects to override these classes in the regular way (b) passing the product to this class means you can alter the form fields based on product (or more likely product class).